README.md

```diff
@@ -1,7 +1,212 @@
 [![Build & Test](https://github.com/trailofbits/vast/actions/workflows/build.yml/badge.svg)](https://github.com/trailofbits/vast/actions/workflows/build.yml)
-# VAST — Verbose AST
+[![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+# VAST: MLIR for Program Analysis
+
+VAST is a library for program analysis and instrumentation of C/C++ and related
+languages. The goal of this tool is to provide a foundation for customizable
+program representation for a broad spectrum of analyses. Using the MLIR
+infrastructure, VAST provides a toolset to represent C/C++ at various stages of
+the compilation and to transform the representation to the best-fit program
+abstraction.
+
+Whether static or dynamic, the program analysis often requires a specific view
+of the source code. The usual requirements for the representation are to be
+easily analyzable, i.e., have a reasonably small set of operations, be truthful
+to the semantics of the analyzed program, and the analysis must be relatable to
+the source. It is also beneficial to access the source at various abstraction
+levels.
+
+The current state-of-the-art tools leverage compiler infrastructures to perform
+program analysis. This approach is beneficial because it remains truthful to the
+executed program semantics, whether AST or LLVM IR. However, these
+representations come at a cost that they are designed for optimization and code
+generation.
+
+The Clang AST is unoptimized and too complex for interpretation-based analysis.
+Also, it lacks program features that Clang inserts during LLVM codegen. On the
+other hand, LLVM is often too low-level and hardly relatable to high-level
+program constructs.
+
+We plan to design a compiler frontend in VAST with program analysis in mind. To
+be specific, VAST will represent the compilation process as a tower of IRs in
+multiple MLIR dialects. The MLIR allows us to capture high-level features from
+AST and interleave them with low-level dialects.
+
+## A Tower of IRs
+
+The feature that differentiates our approach is that the program representation
+can hold multiple representations simultaneously, the so-called `tower of IRs`.
+One can imagine the tower as multiple MLIR modules side-by-side in various
+dialects. Each layer of the tower represents a specific stage of compilation. At
+the top is a high-level dialect relatable to AST, and at the bottom is a
+low-level LLVM-like dialect. Layers are interlinked with location information.
+Higher layers can also be seen as metadata for lower layers.
+
+This feature simplifies analysis build on top of VAST IR in multiple ways. It
+naturally provides __provenance__ to source and higher levels of representation
```

Comment:
```suggestion
naturally provides __provenance__ to higher levels dialects (and source code)
```
Not sure, this was reading as source and then higher levels as source code whereas I think? it was meant as the dialects between AST and LLVM.